planemo/training.py

```diff
@@ -0,0 +1,938 @@
+        raise ValueError("No type for the paramater %s" % tp_desc['name'])
+    if tp_desc['type'] == 'data' or tp_desc['type'] == 'data_collection':
+        paramlist += format_inputs(step_inputs, tp_desc, wf_steps, level)
+    elif tp_desc['type'] == 'section':
+        paramlist += format_section_param_desc(step_params, step_inputs, tp_desc, level, wf_steps)
+    elif tp_desc['type'] == 'conditional':
+        paramlist += format_conditional_param_desc(step_params, step_inputs, tp_desc, level, wf_steps)
+    elif tp_desc['type'] == 'repeat':
+        paramlist += format_repeat_param_desc(step_params, step_inputs, tp_desc, level, wf_steps)
+    else:
+        param_value = get_param_value(step_params, tp_desc, force_default)
+        if param_value is not None:
+            context = {
+                'space': SPACE * level,
+                'param_label': tp_desc['label'],
+                'param_value': param_value}
+            paramlist += templates.render(INPUT_PARAM, **context)
+    return paramlist
+
+
+def get_param_desc(step_params, step_inputs, tp_desc, level, wf_steps, should_be_there=False):
+    """Parse the parameters of the tool and return a formatted list with the values set in the workflow."""
+    paramlist = ''
+    for n, tp_d in tp_desc.items():
+        if n not in step_params:
+            if not should_be_there:
+                info("%s not in workflow" % n)
+            else:
+                raise ValueError("%s not in workflow" % n)
+        else:
+            step_param = get_lower_params(step_params, n)
+            if step_param is None:
+                continue
+            paramlist += format_param_desc(step_param, step_inputs, tp_d, level, wf_steps)
+    return paramlist
+
+
+def get_handson_box(step, steps, tools):
+    """Get the string for an hands-on box based on a step in a workflow."""
+    # get input (if none: input step)
+    step_inputs = get_wf_step_inputs(step['input_connections'])
+    if not step_inputs:
+        return ''
+    # get params
+    step_params = json.loads(step['tool_state'])
+    # get tool
+    tool_name = step['name']
+    tp_desc = tools[tool_name]
+    # get formatted param description
+    paramlist = get_param_desc(step_params, step_inputs, tp_desc, 1, steps, should_be_there=True)
+    context = {"tool_name": tool_name, "paramlist": paramlist}
+    return templates.render(HANDS_ON_TOOL_BOX_TEMPLATE, **context)
+
+
+def create_tutorial_from_workflow(kwds, z_file_links, tuto_dir, ctx):
+    """Create tutorial structure from the workflow file."""
+    # load workflow
+    if kwds['workflow_id']:
+        if not kwds['galaxy_url']:
+            raise ValueError("No Galaxy URL given")
+        if not kwds['galaxy_api_key']:
+            raise ValueError("No API key to access Galaxy given")
+        wf, tools = get_wf_tools_from_running_galaxy(kwds)
+    else:
+        wf, tools = get_wf_tool_from_local_galaxy(kwds, kwds["workflow"], ctx)
+    save_to_yaml(tools, 'tools.yaml')
```

Comment:
Hum, don't know if it's intended but now I get a tools.yaml at the root of the gtn dir (but it doesn't look the same as the tutorial tools.yaml)